plugins/file_chooser/lib/src/channel_controller.dart

```diff
@@ -116,26 +121,18 @@ class FileChooserChannelController {
   static final FileChooserChannelController instance =
       new FileChooserChannelController._();
 
-  /// Shows a file chooser of [type] configured with [options], calling
-  /// [callback] when it completes.
-  void show(FileChooserType type, FileChooserConfigurationOptions options,
-      FileChooserCallback callback) {
-    try {
-      final methodName = type == FileChooserType.open
-          ? _kShowOpenPanelMethod
-          : _kShowSavePanelMethod;
-      _channel
-          .invokeMethod(methodName, options.asInvokeMethodArguments())
-          .then((response) {
-        final paths = response?.cast<String>();
-        final result =
-            paths == null ? FileChooserResult.cancel : FileChooserResult.ok;
-        callback(result, paths);
-      });
-    } on PlatformException catch (e) {
-      print('File chooser plugin failure: ${e.message}');
-    } on Exception catch (e, s) {
-      print('Exception during file chooser operation: $e\n$s');
-    }
+  /// Shows a file chooser of [type] configured with [options], returning a
+  /// [FileChooserResult] when complete.
+  Future<FileChooserResult> show(
+    FileChooserType type,
+    FileChooserConfigurationOptions options,
+  ) async {
+    final methodName = type == FileChooserType.open
+        ? _kShowOpenPanelMethod
+        : _kShowSavePanelMethod;
+    final response = await _channel.invokeMethod(
```

Comment:
Sorry, random drive-by comment, use [MethodChannel.invokeListMethod](https://api.flutter.dev/flutter/services/MethodChannel/invokeListMethod.html)

```suggestion
    final paths = await _channel.invokeListMethod<String>(
```

Reply:
TIL! Thanks for that drive-by :)